src/main/java/nl/b3p/tailormap/api/controller/FeaturesController.java

```diff
@@ -0,0 +1,268 @@
+
+        FeaturesResponse featuresResponse = new FeaturesResponse();
+
+        if (null != filter) {
+            throw new BadRequestException("filter is not currently supported");
+        }
+        if (null != __fid) {
+            throw new BadRequestException("__fid is not currently supported");
+        }
+
+        if (null != x && null != y) {
+            featuresResponse = getFeaturesByXY(appLayer, x, y, distance, simplify);
+        } else {
+            // TODO other implementations
+            throw new BadRequestException("Only x/y/distance/simplify parameters are supported");
+        }
+
+        return featuresResponse;
+    }
+
+    @NotNull
+    private FeaturesResponse getFeaturesByXY(
+            @NotNull ApplicationLayer appLayer,
+            @NotNull Double x,
+            @NotNull Double y,
+            @NotNull Double distance,
+            @NotNull Boolean simplifyGeometry)
+            throws BadRequestException {
+        // validate buffer
+        if (null != distance && 0 > distance) {
+            throw new BadRequestException("Buffer distance must be greater than 0");
+        }
+
+        FeaturesResponse featuresResponse = new FeaturesResponse();
+        // find attribute source of layer
+        GeoService geoService = appLayer.getService();
+        Layer layer = geoService.getLayer(appLayer.getLayerName(), entityManager);
+        SimpleFeatureType sft = layer.getFeatureType();
+        if (null == sft) {
+            return featuresResponse;
+        }
+        List<ConfiguredAttribute> configuredAttributes = appLayer.getAttributes(sft);
+        configuredAttributes =
+                configuredAttributes.stream()
+                        .filter(ConfiguredAttribute::isVisible)
+                        .collect(Collectors.toList());
+
+        try {
+            SimpleFeatureSource fs = FeatureSourceFactoryHelper.openGeoToolsFeatureSource(sft);
+            Query q = new Query(fs.getName().toString());
+            String geomAttribute = fs.getSchema().getGeometryDescriptor().getLocalName();
+            FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+
+            GeometricShapeFactory shapeFact = new GeometricShapeFactory();
+            shapeFact.setNumPoints(32);
+            shapeFact.setCentre(new Coordinate(x, y));
+            shapeFact.setSize(distance * 2d);
+            Polygon p = shapeFact.createCircle();
+            Filter spatialFilter = ff.intersects(ff.property(geomAttribute), ff.literal(p));
+
+            // TODO flamingo does some fancy stuff to combine with existing filters using
+            //      TailormapCQL and some filter visitors
+
+            q.setPropertyNames(
+                    configuredAttributes.stream()
+                            .map(ConfiguredAttribute::getAttributeName)
+                            .collect(Collectors.toList()));
+            q.setFilter(spatialFilter);
+            q.setMaxFeatures(DEFAULT_MAX_FEATURES);
```

Comment:
Not sure if there is a way around it, but setting max features could give unexpected results. If we would be re-using this in an attribute list for example filtering there could really limit the amount of visible rows.

Reply:
we will need to implement a paging mechanism when the time comes: https://b3partners.atlassian.net/browse/HTM-121